cpp/epidemiology/utils/eigen_util.h

```diff
@@ -206,9 +206,11 @@ namespace details
 {
     //true if elements returned by matrix(i, j) are references where matrix is of type M;
     //false if the elements are temporaries, e.g. for expressions like Eigen::MatrixXd::Constant(r, c, v);
+    //Caution: This works only for matrix expressions having the LinearAccessBit:
+    // https://eigen.tuxfamily.org/dox/classEigen_1_1DenseCoeffsBase_3_01Derived_00_01ReadOnlyAccessors_01_4.html#a496672306836589fa04a6ab33cb0cf2a
     template <class M>
     using IsElementReference =
-        std::is_reference<decltype(std::declval<M>()(std::declval<Eigen::Index>(), std::declval<Eigen::Index>()))>;
```

Comment:
I don't know if this now always works as intended.

@dabele Could you check that everyting still is as intended?

Reply:
To be honest, I am not 100% sure why this works with gcc. This check is part of the deserialization of `CustomIndexArray` that stores a flat vector, so MSVC complained that no operator exists accepting two inputs. As long as we only use matrix expressions that can be accessed with the `[]` operator linearly, this check should work for all arrays with contiguous memory.

Reply:
Ok. I'd like to have @dabele checking it after his vacation anyway :)

Reply:
the expression `v(i, j)` should be valid for matrices as well as vectors, I assume to allow this kind of generic programming, see here: https://godbolt.org/z/zqz4qazdq . And I'm pretty sure it's used in other places as well.

the `(i, j)` accessor is more general than the `[i]` accessor, which is why I used that. `RowMajorIterator` is intended to be able to iterate over any matrix or vector expression.

As for why it doesn't work in MSVC, no idea yet.